models/densenet_efficient.py

```diff
@@ -300,16 +300,11 @@ def __init__(self, shared_allocation_1, shared_allocation_2,
         self.momentum = momentum
         self.eps = eps
 
-        # Buffers to store old versions of bn statistics
-        self.prev_running_mean = self.running_mean.new(self.running_mean.size())
-        self.prev_running_var = self.running_var.new(self.running_var.size())
-        self.curr_running_mean = self.running_mean.new(self.running_mean.size())
-        self.curr_running_var = self.running_var.new(self.running_var.size())
-
     def forward(self, bn_weight, bn_bias, *inputs):
-        # Save the current BN statistics for later
-        self.prev_running_mean.copy_(self.running_mean)
-        self.prev_running_var.copy_(self.running_var)
+        if self.training:
+            # Save the current BN statistics for later
+            prev_running_mean = self.running_mean.clone()
+            prev_running_var = self.running_var.clone()
```

Comment:
Rather than calling `.clone` here, it'll be faster if we have a buffer that we copy into. So i'd prefer to keep it the way it currently is.